datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxRepairService.java

```diff
@@ -0,0 +1,157 @@
+package com.sequenceiq.datalake.service.sdx;
+
+import static com.sequenceiq.cloudbreak.exception.NotFoundException.notFound;
+
+import java.util.Collections;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.ws.rs.ClientErrorException;
+import javax.ws.rs.NotFoundException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.dyngr.Polling;
+import com.dyngr.core.AttemptResult;
+import com.dyngr.core.AttemptResults;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.Status;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.ClusterRepairV4Request;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.StackV4Response;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.cluster.ClusterV4Response;
+import com.sequenceiq.cloudbreak.auth.altus.Crn;
+import com.sequenceiq.cloudbreak.auth.altus.CrnParseException;
+import com.sequenceiq.cloudbreak.client.CloudbreakServiceUserCrnClient;
+import com.sequenceiq.cloudbreak.common.exception.ClientErrorExceptionHandler;
+import com.sequenceiq.cloudbreak.common.json.JsonUtil;
+import com.sequenceiq.datalake.controller.exception.BadRequestException;
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.entity.SdxClusterStatus;
+import com.sequenceiq.datalake.flow.SdxReactorFlowManager;
+import com.sequenceiq.datalake.repository.SdxClusterRepository;
+import com.sequenceiq.sdx.api.model.SdxClusterRequest;
+import com.sequenceiq.sdx.api.model.SdxRepairRequest;
+
+@Service
+public class SdxRepairService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SdxRepairService.class);
+
+    @Inject
+    private SdxClusterRepository sdxClusterRepository;
+
+    @Inject
+    private SdxReactorFlowManager sdxReactorFlowManager;
+
+    @Inject
+    private CloudbreakServiceUserCrnClient cloudbreakClient;
+
+    @Inject
+    private SdxService sdxService;
+
+    public void triggerRepair(String userCrn, String clusterCrn, SdxRepairRequest clusterRepairRequest) {
+        SdxCluster cluster = sdxService.getByCrn(userCrn, clusterCrn);
```

Comment:
please add cluster object to MDCContext, then every field in MDCContext should be filled: `MDCBuilder.buildMdcContext(sdxCluster);`